web/html/src/manager/systems/virtual-list.tsx

```diff
@@ -0,0 +1,143 @@
+
+// See java/code/src/com/suse/manager/webui/templates/systems/virtual-list.jade
+type Props = {
+  /** Locale of the help links */
+  docsLocale: string;
+  isAdmin: boolean;
+};
+
+function VirtualSystems(props: Props) {
+  const [selectedSystems, setSelectedSystems] = useState([]);
+  const [selectedSystemsCount, setCount] = useState(0);
+
+  const handleSelectedSystems = (data) => {
+    setSelectedSystems(data);
+  };
+
+  useEffect(()=> {
+    setCount(selectedSystems.length);
+    document.getElementById("header_selcount").innerHTML =
+      "<span id='spacewalk-set-system_list-counter' class='badge'>" +
+      selectedSystemsCount.toString() +
+      "</span>" +
+      (selectedSystemsCount === 1 ? "system selected" : "systems selected");
+  }, [handleSelectedSystems]);
+
+  const addToSSM = () => {
+    var url = "/rhn/manager/systems/addToSsm";
+    Network.post(url, selectedSystems);
+  };
+
+  const searchData = (datum, criteria) => {
+    if (criteria) {
+      return datum.name.toLowerCase().includes(criteria.toLowerCase());
+    }
+    return true;
+  };
+
+  return (
+    <>
+      <h1>
+        <IconTag type="header-system" />
+        {t("Virtual Systems")}
+        <a
+          href={`/docs/${props.docsLocale}/reference/systems/systems-list.html`}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <IconTag type="header-help" />
+        </a>
+      </h1>
+
+      <div>
+        <button className="btn btn-default" onClick={addToSSM}>
+          {t("Add Selected to SSM")}
+        </button>
+      </div>
+
+      <Table
+        data="/rhn/manager/api/systems/list/virtual"
+        identifier={(item) => item.uuid}
+        initialSortColumnKey="hostServerName"
+        selectable
+        isSelectEnabled={(item) => item.hasOwnProperty("virtualSystemId")}
+        selectedItems={selectedSystems}
+        onSelect={handleSelectedSystems}
+        initialItemsPerPage={window.userPrefPageSize}
+        searchField={<SearchField filter={searchData} placeholder={t("Filter by System name: ")} />}
+        emptyText={t("No Virtual Systems.")}
+      >
+        <Column
+          columnKey="hostServerName"
+          comparator={Utils.sortByText}
+          header={t("Virtual Host")}
+          cell={(item) => {
+            return <a href={`/rhn/systems/details/Overview.do?sid=${item.hostSystemId}`}>{item.hostServerName}</a>;
+          }}
+        />
+        <Column
+          columnKey="name"
+          comparator={Utils.sortByText}
+          header={t("Virtual System")}
+          cell={(item) => {
+            if (item.systemId != null) {
+              return <a href={`/rhn/systems/details/Overview.do?sid=${item.systemId}`}>{item.name}</a>;
+            }
+            return item.name;
+          }}
+        />
+        <Column
+          columnKey="stateName"
+          comparator={Utils.sortByText}
+          header={t("Status")}
+          cell={(items) => items.stateName}
```

Comment:
Minor inconsistent naming, other columns refer to `item` in their cell definition, here it's suddenly `items`. Would be nice to be consistent to avoid possible confusion.

Reply:
I fixed some of them, looks like I overlooked others